remote-api/src/main/java/org/alfresco/rest/api/model/ActionParameterConstraint.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2021 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.api.model;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.alfresco.service.Experimental;
+
+/**
+ * Representation of action parameter constraint.
```

Comment:
If I've understood correctly then this is just a specific type of constraint. It's basically where there is a list of possible values - like an enum.  If this is right then I think it would be useful to explain a bit more about it here.

Reply:
I've extended the javadoc to be (hopefully) more descriptive.